### ISSUE TITLE
Focus issues with tabs on dialog component

### ISSUE THREAD

**Issue by GCHQ-Developer-847** [mi6/ic-ui-kit#4246]
### Summary of the bug 
When tabs are slotted into a dialog component, you can tab forwards fine through all the focusable elements but if you try pressing Shift + Tab to move focus backwards, the focus gets stuck on the tabs.

Also if you change the selected tab, the first tab stays focusable but it shouldn't be anymore (only the selected tab should be focusable).

The same issue happens on the drawer component. See "Additional information" section below.

### 🪜 How to reproduce 
1. Go to this [StackBlitz](https://stackblitz.com/edit/6napuydj?file=src%2Fapp.tsx&hideExplorer=1) and open the dialog.
2. Try tabbing forward - all works fine, but if you tab backwards (Shift + Tab), the focus gets stuck on the tabs.

Also:
1. Change the selected tab.
2. Try tabbing through the focusable elements and see that the first tab remains focusable.

### 🖥 📱 Device 
- Tested on Chrome

### 🧐 Expected behaviour 
Shift + Tab should make the focus cycle backwards through all focusable elements on the dialog and not get stuck on the tabs. And only the selected tab should be focusable.

### Additional info 
The Shift + Tab issue only seems to happen if the first tab is selected.

**Note:** To avoid conflicts, this ticket should ideally be worked on after the drawer component has been merged. Please check that the solution fixes the same issue when tabs are used in a drawer as it uses the same functionality (i.e. if you need to make a change to the helper functions the dialog uses, make the same change in the canary helpers.ts file). 

Please uncomment the drawer Cypress test "should focus interactive content and trap focus - tabs" when this issue has been fixed.

**Comment by GCHQ-Developer-530** [comment]
I've noticed that when you change the selected tab and then use Shift + Tab, then it does allow you to cycle through the slotted components.
